runconfig/hostconfig_test.go

```diff
@@ -60,16 +60,30 @@ func TestNetworkModeTest(t *testing.T) {
 	}
 }
 
+func TestDaemonDefaultIpcMode(t *testing.T) {
+	dm := DefaultDaemonIpcMode()
+	if dm == "" {
+		t.Fatalf("DefaultDaemonIpcMode should not be empty")
+	}
+	if !dm.Valid() {
+		t.Fatalf("DefaultDaemonIpcMode (%v) is not valid", dm)
+	}
+	if !dm.IsShareable() && !dm.IsPrivate() {
+		t.Fatalf("DefaultDaemonIpcMode (%v) does not make sense", dm)
+	}
+}
+
 func TestIpcModeTest(t *testing.T) {
 	ipcModes := map[container.IpcMode][]bool{
-		// private, host, container, valid
-		"":                         {true, false, false, true},
-		"something:weird":          {true, false, false, false},
-		":weird":                   {true, false, false, true},
-		"host":                     {false, true, false, true},
-		"container:name":           {false, false, true, true},
-		"container:name:something": {false, false, true, false},
-		"container:":               {false, false, true, false},
+		// private, host, container, shareable, valid
+		"":                {false, false, false, false, true},
```

Comment:
nit: would it be more readable to use a struct for these? The second loop for parsing container names could be combined in that case;

```go
func TestIpcModeTest(t *testing.T) {
	ipcModes := map[container.IpcMode]struct {
		private   bool
		host      bool
		container bool
		shareable bool
		valid     bool
		ctrName   string
	}{
		"":                      {valid: true},
		"private":               {private: true, valid: true},
		"something":             {},
		"something:weird":       {ctrName: "weird"},
		":weird":                {ctrName: "weird"},
		"host":                  {host: true, valid: true},
		"container":             {},
		"container:name":        {container: true, ctrName: "name", valid: true},
		"container:name1:name2": {container: true, ctrName: "name1:name2", valid: true},
		"container:":            {container: true, valid: true},
		"shareable":             {shareable: true, valid: true},
	}
	for ipcMode, state := range ipcModes {
		assert.Equal(t, state.private, ipcMode.IsPrivate(), "IpcMode.IsPrivate() parsing failed for %q", ipcMode)
		assert.Equal(t, state.host, ipcMode.IsHost(), "IpcMode.IsHost()  parsing failed for %q", ipcMode)
		assert.Equal(t, state.container, ipcMode.IsContainer(), "IpcMode.IsContainer()  parsing failed for %q", ipcMode)
		assert.Equal(t, state.shareable, ipcMode.IsShareable(), "IpcMode.IsShareable()  parsing failed for %q", ipcMode)
		assert.Equal(t, state.valid, ipcMode.Valid(), "IpcMode.Valid()  parsing failed for %q", ipcMode)
		assert.Equal(t, state.ctrName, ipcMode.Container(), "IpcMode.Container() parsing failed for %q", ipcMode)
	}
}
```